ninja_cursor_pagination/__init__.py

```diff
@@ -76,7 +76,7 @@ def decode_cursor(cls, encoded_cursor: str | None) -> Cursor:
 
     class Output(Schema):
         results: list[Any] = Field(description=_("The page of objects."))
-        count: int = Field(
+        count: int | None = Field(
```

Comment:
This isn't ideal, since by default, the schema should indicate that simply an `int` is always returned.

This type information (whether or not `None` is possible) is knowable at the time that the endpoint is defined and it is consistent across all usages of a given endpoint.

Reply:
I tried using [`create_model`](https://docs.pydantic.dev/latest/concepts/models/#dynamic-model-creation) to patch an altered `Output` model in the constructor. This seems to work functionally, but it doesn't result in the correct OpenAPI schema, which is the whole point of trying to fix this issue.

The problem is that for all endpoints returning `list[CategorySchema]`, Ninja tries to define a `"PagedCategorySchema"` in OpenAPI. If different endpoints use different types of pagination, the different contents of `PagedCategorySchema` silently overwrite each other.